### PR TITLE
Separate frontend to s3 host for tm4 deployment

### LINKED ISSUE #2445: Solve create-react-app in a cloud deployment
When testing TM4 on a mapathon the following problem has been detected: A subset of users received a whitescreen, either directly after authentication of when browsing to a project. The error was identified as `GET https://tm4.hotosm.org/static/js/6.da8f26eb.chunk.js net::ERR_ABORTED 404`.

@willemarcel has suggested two possible approaches to solve this:

> * Serve it from S3 instead of having a flask view: https://medium.com/@wolovim/deploying-create-react-app-to-s3-or-cloudfront-48dae4ce0af
> * Push the frontend release build to a GitHub Branch or a S3 bucket on the build pipeline, download it when deploying new instances and use the build instead of building it before starting the server

### PR DESCRIPTION
**Pull request by dakotabenjamin**
Closes #2445 

This PR affects the deployment of Tasking Manager by setting up a single frontend build hosted on AWS S3 and using an AWS Cloudfront distribution. We add new CloudFormation resources for those AWS items, and we have moved the environment variables for frontend and server to their respective deployments. 

Currently you can view the deployment working here: https://d1qxwg0plztens.cloudfront.net. I will need to remove the deployment from the CircleCI config before merging, however. 

Tasks left to complete:

- [x] Add route53 record for backend url
- [x] test adding project and locking/unlocking tasks
- [ ] Remove the flask views that serve the frontend (@willemarcel)
- [ ] Update the documentation on how to setup locally and deploy TM after this change
- [x] Remove temp deployment from circleci code

### CODE REVIEW

**Comment by xamanu** [comment]
Please squash and merge. What is the best way for testing this?

File: .circleci/config.yml [hunk 1]
```diff
@@ -232,6 +232,39 @@ jobs:
...
+          export TM_ORG_NAME=${<< parameters.org_name >>}
+          export TM_ORG_CODE=${<< parameters.org_code >>}
+          export TM_ORG_URL=${<< parameters.org_url >>}
+          export TM_ORG_PRIVACY_POLICY=${<< parameters.org_privacy_policy >>}
+          export TM_ORG_TWITTER=${<< parameters.org_twitter >>}
+          export TM_ORG_FB=${<< parameters.org_fb >>}
+          export TM_ORG_INSTAGRAM=${<< parameters.org_instagram >>}
+          export TM_ORG_YOUTUBE=${<< parameters.org_youtube >>}
+          export TM_ORG_GITHUB=${<< parameters.org_github >>}
+          export TM_APP_API_URL="https://`aws cloudformation list-exports --query "Exports[?Name=='tasking-manager-<< parameters.stack_name >>-loadbalancer-url-${AWS_REGION}'].Value" --output=text`"
```
Comment (willemarcel) [inline .circleci/config.yml]: @dakotabenjamin could you explain your intention on that part? Will you change it later to the public URL (tasks-stage / tm4 / tasks.hotosm.org)?

**Comment by willemarcel** [comment]
Some itens I think we could work on that PR (I can help with them):

- Remove the flask views that serves the frontend
- Update the documentation on how to setup locally and deploy TM after this change

File: .circleci/config.yml (hunk 1 above)
Comment (dakotabenjamin) [reply .circleci/config.yml]: The frontend uses the public facing URL, not the server. 
Tasking Manager(1)

**Comment by dakotabenjamin** [comment]
@xamanu we may need to add some dummy projects to the testing site to make sure everything works correctly. I do not know how to change user permissions on there though.

File: .circleci/config.yml (hunk 1 above)
Comment (willemarcel) [reply .circleci/config.yml]: @dakotabenjamin If I would like to use the TM API on another application. Which URL should I use?

**Comment by dakotabenjamin** [comment]
@xamanu @willemarcel For the portions of this PR that are critical to the mapathon on thursday, we are ready to review. The test deployment is https://d1qxwg0plztens.cloudfront.net and is running well. Please review and merge as soon as you are able. 

After merging, I need to change the target alias for tm4.hotosm.org to the frontend cloudfront distribution and we will be complete.

**Comment by xamanu** [comment]
It looks good so far. How do we coordinate the deployment?

**Comment by dakotabenjamin** [comment]
Let's merge to develop, test things there, and then merge to tm4. I can handle the AWS needs if you do the merging and deployment to tm4

**Pull request merged**